packages/events/EventPluginHub.js

```diff
@@ -163,17 +163,27 @@ function extractEvents(
   targetInst: null | Fiber,
   nativeEvent: AnyNativeEvent,
   nativeEventTarget: EventTarget,
+  passive: null | boolean,
 ): Array<ReactSyntheticEvent> | ReactSyntheticEvent | null {
   let events = null;
   for (let i = 0; i < plugins.length; i++) {
     // Not every plugin in the ordering may be loaded at runtime.
     const possiblePlugin: PluginModule<AnyNativeEvent> = plugins[i];
     if (possiblePlugin) {
+      // Check if the plugin supports legacy or non legacy events
+      // based on the passive flag being null or a boolean
+      if (
+        (possiblePlugin.isLegacy && passive !== null) ||
```

Comment:
To avoid checking every plugin every time, can we have two lists of plugins, and switch early depending on the event?